tests/data_classes.py

```diff
@@ -0,0 +1,157 @@
+            pass
+
+
+class Resource:
+    DEFAULT_JSON_FILE = 'config.json'
+
+    def __init__(self):
+        self.zfs = ZFS()
+
+
+# A simple class which can tell us the state of the jail
+class Jail(Resource):
+
+    def __init__(self, name):
+        super().__init__()
+        self.name = name
+
+    @property
+    def path(self):
+        # Jail can be either under `jails` or `templates` datasets
+        if self.zfs.get_dataset(
+                f'{self.zfs.pool.name}/iocage/jails/{self.name}'
+        ):
+            dataset = 'jails'
+        else:
+            dataset = 'templates'
+        return f'{self.zfs.pool.name}/iocage/{dataset}/{self.name}'
+
+    @property
+    def exists(self):
+        return self.zfs.get_dataset(self.path) is not None
+
+    @property
+    def template(self):
+        return 'iocage/templates/' in self.path
+
+    @property
+    def jid(self):
+        try:
+            return int(
+                subprocess.check_output(
+                    ['jls', '-j', f'ioc-{self.name.replace(".", "_")}'],
+                    stderr=subprocess.PIPE
+                ).decode('utf-8').split()[5]
+            )
+        except subprocess.CalledProcessError:
+            return None
+
+    @property
+    def running(self):
+        return self.jid is not None
+
+    @property
+    def config(self):
+        if os.path.exists(os.path.join(self.path, 'config.json')):
+            with open(os.path.join(self.path, 'config.json'), 'r') as f:
+                return json.loads(f.read())
+        else:
+            return {}
+
+    @property
+    def thick_config_jail(self):
+        return 'CONFIG_TYPE' in self.config
+
+    @property
+    def base_jail(self):
+        return self.config.get('base_jail', 'no') == 'yes'
+
+
+class JailSelector:
+
+    def __init__(self):
+        self.zfs = ZFS()
+
+    @property
+    def iocage_dataset(self):
+        return self.zfs.get_dataset(f'{self.zfs.pool.name}/iocage')
+
+    @property
+    def jails_dataset(self):
+        return self.zfs.get_dataset(f'{self.iocage_dataset.name}/jails')
+
+    @property
+    def templates_dataset(self):
+        return self.zfs.get_dataset(f'{self.iocage_dataset.name}/templates')
+
+    @property
+    def all_jails(self):
+        return [
+            Jail(d.name.split('/')[-1])
+            for d in itertools.chain(
+                self.jails_dataset.children, self.templates_dataset.children
+            )
+        ]
+
+    @property
+    def jails(self):
+        return [j for j in self.all_jails if not j.template]
+
+    @property
+    def running_jails(self):
+        return [j for j in self.all_jails if j.running]
+
+    @property
+    def stopped_jails(self):
+        return [j for j in self.all_jails if not j.running]
+
+    @property
+    def template_jails(self):
+        return [j for j in self.all_jails if j.template]
+
+    @property
+    def thick_config_jails(self):
+        return [j for j in self.all_jails if j.thick_config_jail]
```

Comment:
In addition to the suggestion on the definition above:
```suggestion
        return [j for j in self.all_jails if j.is_thickconfig]
```